tests/phpunit/tests/rest-api/rest-block-type-controller.php

```diff
@@ -0,0 +1,496 @@
+				'label'        => 'Blue Quote',
+				'style_handle' => 'myguten-style',
+			),
+		);
+		$this->assertEqualSets( $expected, $data['styles'] );
+
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_block_invalid_name() {
+		$block_type = 'fake/block';
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-types/' . $block_type );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_block_type_invalid', $response, 404 );
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_item_invalid() {
+		$block_type = 'fake/invalid';
+		$settings   = array(
+			'title'            => true,
+			'description'      => true,
+			'icon'             => true,
+			'attributes'       => 'invalid_attributes',
+			'provides_context' => 'invalid_provides_context',
+			'uses_context'     => 'invalid_uses_context',
+			'category'         => true,
+			'editor_script'    => true,
+			'script'           => true,
+			'editor_style'     => true,
+			'style'            => true,
+			'keywords'         => 'invalid_keywords',
+			'example'          => 'invalid_example',
+			'parent'           => 'invalid_parent',
+			'supports'         => 'invalid_supports',
+			'styles'           => 'invalid_styles',
+			'render_callback'  => 'invalid_callback',
+			'textdomain'       => true,
+		);
+		register_block_type( $block_type, $settings );
+		wp_set_current_user( self::$admin_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/block-types/' . $block_type );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals( $block_type, $data['name'] );
+		$this->assertEquals( '1', $data['title'] );
+		$this->assertEquals( '1', $data['description'] );
+		$this->assertEquals( null, $data['icon'] );
+		$this->assertEquals( null, $data['editor_script'] );
+		$this->assertEquals( null, $data['script'] );
+		$this->assertEquals( null, $data['editor_style'] );
+		$this->assertEquals( null, $data['style'] );
+		$this->assertEqualSets( array(), $data['provides_context'] );
+		$this->assertEqualSets(
+			array(
+				'layout' => array(
+					'type' => 'string',
+				),
+			),
+			$data['attributes']
+		);
+		$this->assertEqualSets( array( 'invalid_uses_context' ), $data['uses_context'] );
+		$this->assertEqualSets( array( 'invalid_keywords' ), $data['keywords'] );
+		$this->assertEqualSets( array( 'invalid_parent' ), $data['parent'] );
+		$this->assertEqualSets( array(), $data['supports'] );
+		$this->assertEqualSets( array( 'invalid_styles' ), $data['styles'] );
+		$this->assertEquals( null, $data['example'] );
+		$this->assertEquals( null, $data['category'] );
+		$this->assertEquals( null, $data['textdomain'] );
+		$this->assertFalse( false, $data['is_dynamic'] );
+	}
+
+	/**
+	 * @ticket 47620
+	 */
+	public function test_get_item_defaults() {
+		$block_type = 'fake/false';
+		$settings   = array(
```

Comment:
Default values test @TimothyBJacobs

Reply:
Default tests look good, thanks for adding them!